Segmentation fault TPUART

Hello,

I got segmentation fault when I launch knxd on tpuarts, any idea ?

knxd -t 0xffff -e 2.1.250 -E 2.1.251:4 -u /tmp/knxd_2 tpuarts:/dev/ttyS2

Layer 4 [ 1:main        0.000] initialized
Layer 4 [ 1:main        0.000] setting up
Layer 3 [ 4:A.unix/local       0.000] registerLink: 4:A.unix
Layer 0 [11:B.tpuarts/log       0.000] Insert 10:LowF / 9:tpuart
Layer 0 [11:B.tpuarts/log       0.000] Setup
Layer 0 [13:B.tpuarts/log       0.000] Insert 12:TPU_ser / 10:LowF
Layer 0 [11:B.tpuarts/log       0.000] Filter single not found
Layer 0 [13:B.tpuarts/log       0.000] Setup
Layer 0 [13:B.tpuarts/log       0.000] Setup OK: yes
Layer 0 [11:B.tpuarts/log       0.000] Setup OK: yes
Layer 3 [ 8:B.tpuarts/Conn      0.000] registerLink: 8:B.tpuarts
Layer 4 [ 1:main                0.000] setup OK
Layer 4 [ 1:main                0.000] trigger going up
Layer 3 [ 8:B.tpuarts/Conn      0.000] Start: cfg:B.tpuarts
Layer 5 [ 8:B.tpuarts/Conn      0.001] down => >up
Layer 5 [ 8:B.tpuarts/Conn      0.001] Starting
Layer 0 [11:B.tpuarts/log       0.001] Start
Layer 0 [13:B.tpuarts/log       0.001] Start
Layer 0 [12:B.tpuarts/TPU_ser   0.002] Opened /dev/ttyS2 with baud 19200
Layer 2 [12:B.tpuarts/TPU_ser   0.002] Opened
Layer 2 [12:B.tpuarts/TPU_ser   0.002] Buffer Setup on fd 4
Layer 0 [13:B.tpuarts/log       0.002] Started
Layer 8 [10:B.tpuarts/LowF      0.002] state: new > start
Layer 0 [10:B.tpuarts/LowF      0.002] SendReset 01
Layer 0 [13:B.tpuarts/log       0.002] Send(001): 01
Layer 4 [ 8:B.tpuarts/Conn      0.002] link state changed: >up
Layer 3 [ 4:A.unix/local        0.002] Start: cfg:A.unix
Layer 5 [ 4:A.unix/local        0.002] down => >up
Layer 8 [ 4:A.unix/local        0.002] OpenLocalSocket /tmp/knxd_2
Layer 8 [ 4:A.unix/local        0.003] LocalSocket opened
Layer 5 [ 4:A.unix/local        0.003] >up => up
Layer 4 [ 4:A.unix/local        0.003] link state changed: up
Layer 5 [ 4:A.unix/local        0.003] Started
Layer 4 [ 4:A.unix/local        0.003] link state changed: up
Layer 4 [ 1:main                0.003] going up triggered
Layer 4 [ 1:main                0.003] check start
Layer 4 [ 8:B.tpuarts/Conn      0.003] state is >up
Layer 4 [ 1:main                0.003] check end: want_up 1 some 1>1 all 0>0, going 1 up 1 down 0
Layer 0 [13:B.tpuarts/log       0.004] Recv(001): C0
Segmentation fault

@dallmair cannot upgrade it .... 

Error during compiilation..

```
dh_systemd_start: error: Cannot open(knxnet.socket) for extracting the Also= line(s): No such file or directory
make[1]: Leaving directory '/install/knxd.install2/knxd'
   dh_installdocs
        install -m0755 -d debian/knxd/usr/share/doc/knxd
        install -m0755 -d debian/knxd/usr/share/doc/knxd
        cp --reflink=auto -a ./doc/inifile.rst debian/knxd/usr/share/doc/knxd
        cp --reflink=auto -a ./AUTHORS debian/knxd/usr/share/doc/knxd
        cp --reflink=auto -a ./README.md debian/knxd/usr/share/doc/knxd
        chown -R 0:0 debian/knxd/usr/share/doc
        chmod -R u\+rw,go=rX debian/knxd/usr/share/doc
        install -p -m0644 debian/copyright debian/knxd/usr/share/doc/knxd/copyright
        install -m0755 -d debian/knxd-tools/usr/share/doc/knxd-tools
        install -p -m0644 debian/copyright debian/knxd-tools/usr/share/doc/knxd-tools/copyright
        install -m0755 -d debian/knxd-dev/usr/share/doc/knxd-dev
        install -p -m0644 debian/copyright debian/knxd-dev/usr/share/doc/knxd-dev/copyright
   dh_installchangelogs
        install -m0755 -d debian/knxd/usr/share/doc/knxd
        install -m0755 -d debian/knxd-tools/usr/share/doc/knxd-tools
        install -m0755 -d debian/knxd-dev/usr/share/doc/knxd-dev
        install -p -m0644 debian/.debhelper/generated/knxd-tools/dh_installchangelogs.dch.trimmed debian/knxd-tools/u                                                                                                                        sr/share/doc/knxd-tools/changelog.Debian
        install -p -m0644 debian/.debhelper/generated/knxd/dh_installchangelogs.dch.trimmed debian/knxd/usr/share/doc                                                                                                                        /knxd/changelog.Debian
        install -p -m0644 debian/NEWS debian/knxd-tools/usr/share/doc/knxd-tools/NEWS.Debian
        install -p -m0644 debian/NEWS debian/knxd/usr/share/doc/knxd/NEWS.Debian
        install -p -m0644 debian/.debhelper/generated/knxd-dev/dh_installchangelogs.dch.trimmed debian/knxd-dev/usr/s                                                                                                                        hare/doc/knxd-dev/changelog.Debian
        install -p -m0644 ./ChangeLog debian/knxd-tools/usr/share/doc/knxd-tools/changelog
        install -p -m0644 ./ChangeLog debian/knxd/usr/share/doc/knxd/changelog
        install -p -m0644 debian/NEWS debian/knxd-dev/usr/share/doc/knxd-dev/NEWS.Debian
        install -p -m0644 ./ChangeLog debian/knxd-dev/usr/share/doc/knxd-dev/changelog
   dh_systemd_enable
        install -m0755 -d debian/knxd/lib/systemd/system
        install -p -m0644 debian/knxd.service debian/knxd/lib/systemd/system/knxd.service
        install -m0755 -d debian/knxd/lib/systemd/system
        install -p -m0644 debian/knxd.socket debian/knxd/lib/systemd/system/knxd.socket
        [META] Append autosnippet "postinst-systemd-enable" to postinst [debian/knxd.postinst.debhelper]
        [META] Append autosnippet "postinst-systemd-enable" to postinst [debian/knxd.postinst.debhelper]
        [META] Prepend autosnippet "postrm-systemd" to postrm [debian/knxd.postrm.debhelper.new]
        mv debian/knxd.postrm.debhelper.new debian/knxd.postrm.debhelper
   dh_installinit
        install -m0755 -d debian/knxd/lib/systemd/system
        install -p -m0644 debian/knxd.service debian/knxd/lib/systemd/system/knxd.service
        install -m0755 -d debian/knxd/etc/default
        install -p -m0644 debian/knxd.default debian/knxd/etc/default/knxd
        install -m0755 -d debian/knxd/etc/init.d
        install -p -m0755 debian/knxd.init debian/knxd/etc/init.d/knxd
        [META] Append autosnippet "preinst-init-chmod" to preinst [debian/.debhelper/generated/knxd/preinst.service]
        [META] Append autosnippet "postinst-init-restart" to postinst [debian/.debhelper/generated/knxd/postinst.serv                                                                                                                        ice]
        [META] Append autosnippet "prerm-init-norestart" to prerm [debian/.debhelper/generated/knxd/prerm.service]
        [META] Append autosnippet "postrm-init" to postrm [debian/.debhelper/generated/knxd/postrm.service]
   dh_systemd_start
dh_systemd_start: warning: Could not find "knxnet.socket" in the /lib/systemd/system directory of knxd. This could be                                                                                                                         a typo, or using Also= with a service file from another package. Please check carefully that this message is harmles                                                                                                                        s.
dh_systemd_start: error: Cannot open(knxnet.socket) for extracting the Also= line(s): No such file or directory
make: *** [debian/rules:13: binary] Error 255
dpkg-buildpackage: error: debian/rules binary subprocess returned exit status 2
```

Thanks looks better but still unable to install it:

```
cp systemd/knxd.udev debian/knxd.udev
dh_install
dh_install: warning: Cannot find (any matches for) "usr/lib/systemd/system/knxd-net.socket" (tried in ., debian/tmp)

dh_install: warning: knxd missing files: usr/lib/systemd/system/knxd-net.socket
        install -m0755 -d debian/knxd//etc
        cp --reflink=auto -a debian/tmp/etc/knxd.conf debian/knxd//etc/
        install -m0755 -d debian/knxd//usr/bin
        cp --reflink=auto -a debian/tmp/usr/bin/knxd debian/knxd//usr/bin/
        install -m0755 -d debian/knxd//usr/lib
        cp --reflink=auto -a debian/tmp/usr/lib/knxd_args debian/knxd//usr/lib/
dh_install: error: missing files, aborting
make[1]: *** [debian/rules:27: override_dh_install] Error 255
make[1]: Leaving directory '/install/knxd.install2/knxd'
make: *** [debian/rules:13: binary] Error 2
dpkg-buildpackage: error: debian/rules binary subprocess returned exit status 2
```

--- a/src/backend/tpuart.cpp
+++ b/src/backend/tpuart.cpp
@@ -548,7 +548,7 @@ TPUARTwrap::recv_Data(CArray &c)
         */
       else if (c == 0xCC || c == 0xC0 || c == 0x0C)
         {
-          RecvLPDU (in.data(), 1);
+          RecvLPDU (&c, 1);
         }
       else if ((c & 0x50) == 0x10) // Matches KNX control byte L_Data_Standard/Extended Frame
         {

